app/preload/accountReport.ts

```diff
@@ -8,16 +8,75 @@ import {
     TransactionStatus,
     TransferTransactionWithNames,
     TransactionFilter,
+    TransactionOrder,
+    Global,
+    CredentialNumberPrfKey,
 } from '~/utils/types';
 import exportTransactionFields from '~/constants/exportTransactionFields.json';
 import { getISOFormat } from '~/utils/timeHelpers';
 import transactionKindNames from '~/constants/transactionKindNames.json';
-import transactionMethods from './database/transactionsDao';
 import { AccountReportMethods } from './preloadTypes';
-import { getActiveBooleanFilters } from '~/utils/accountHelpers';
 import { isShieldedBalanceTransaction } from '~/utils/transactionHelpers';
 import AbortController from '~/utils/AbortController';
 import { getEntryName } from './database/addressBookDao';
+import { convertIncomingTransaction } from '~/utils/TransactionConverters';
+import httpMethods from './http';
+import decryptAmountsDao from './database/decryptedAmountsDao';
+import decryptTransactions from '~/utils/decryptHelpers';
+
+async function enrichWithDecryptedAmounts(
+    credentialNumber: number,
+    prfKeySeed: string,
+    address: string,
+    global: Global,
+    transactions: TransferTransaction[]
+): Promise<TransferTransaction[]> {
+    const encryptedTypes = [
+        TransactionKindString.EncryptedAmountTransfer,
+        TransactionKindString.EncryptedAmountTransferWithMemo,
+    ];
+
+    const encryptedTransactions = transactions.filter((t) =>
+        encryptedTypes.includes(t.transactionKind)
+    );
+    const decryptedAmounts = await decryptAmountsDao.findEntries(
+        encryptedTransactions.map(
+            (encryptedTransaction) => encryptedTransaction.transactionHash
+        )
+    );
+
+    const withDecryptedAmounts: TransferTransaction[] = [];
+
+    for (const t of transactions) {
+        if (!encryptedTypes.includes(t.transactionKind)) {
+            withDecryptedAmounts.push(t);
+        } else {
+            const amount = decryptedAmounts.find(
+                (decrypted) => decrypted.transactionHash === t.transactionHash
+            )?.amount;
+            if (amount) {
+                withDecryptedAmounts.push({ ...t, decryptedAmount: amount });
+            } else {
+                const decryptedTransaction = (
+                    await decryptTransactions(
```

Comment:
In this function, it says that there is a "significant performance hit if we decrypt one at a time", and yet we do that here. Was the performance hit not that significant, or should we decrypt these together?